Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -702,70 +702,77 @@ static RefPtr<CSSValue> consumeFontVariantEastAsian(CSSParserTokenRange& range)
     if (range.peek().id() == CSSValueNormal)
         return consumeIdent(range);
     
-    RefPtr<CSSValueList> values = CSSValueList::createSpaceSeparated();
     std::optional<FontVariantEastAsianVariant> variant;
     std::optional<FontVariantEastAsianWidth> width;
     std::optional<FontVariantEastAsianRuby> ruby;
     
-    while (!range.atEnd()) {
-        if (range.peek().type() != IdentToken)
-            return nullptr;
+    auto parseSomethingWithoutError = [&range, &variant, &width, &ruby] () {
+        bool hasParsedSomething = false;
```

Comment:
Mostly because I'm not a huge fan of `parseSomethingWithoutError` as name, I would suggest doing something like `while (!shouldFail)` and set `shouldFail` appropriately, then add early `return nullptr` if `shouldFail` is true.

Reply:
I think it would make the workflow more complicated (and kinda defeat the purpose of the lambda).